modules/l4fail2ban/matcher.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2024 SICK AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4fail2ban
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+
+	caddy_fail2ban "github.com/Javex/caddy-fail2ban"
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(&Fail2Ban{})
+}
+
+type Fail2Ban struct {
+	Banfile string `json:"banfile"`
```

Comment:
It's more idiomatic in Go to have Pascal case for struct field and snake case for json fields

```suggestion
	BanFile string `json:"ban_file"`
```